nemo_skills/inference/server/model.py

```diff
@@ -552,16 +553,28 @@ def __init__(
             base_url = os.getenv("NEMO_SKILLS_OPENAI_BASE_URL", f"http://{host}:{port}/v1")
 
         if api_key is None:
-            if base_url is not None and 'api.nvidia.com' in base_url:
-                api_key = os.getenv("NVIDIA_API_KEY", api_key)
-                if not api_key:
-                    raise ValueError("NVIDIA_API_KEY is required for Nvidia-hosted models.")
+            if base_url is not None and 'nvidia.com' in base_url:
+                if endpoint_type == "azureopenai":
+                    api_key = os.getenv("AZURE_OPENAI_API_KEY", api_key)
+                    if not api_key:
+                        raise ValueError("AZURE_OPENAI_API_KEY is required for Azure OpenAI models.")
+                elif endpoint_type == "openai":
+                    api_key = os.getenv("NVIDIA_API_KEY", api_key)
+                    if not api_key:
+                        raise ValueError("NVIDIA_API_KEY is required for Nvidia-hosted models.")
             elif base_url is not None and 'api.openai.com' in base_url:
                 api_key = os.getenv("OPENAI_API_KEY", api_key)
                 if not api_key:
                     raise ValueError("OPENAI_API_KEY is required for OpenAI models.")
 
-        self.client = OpenAI(api_key=api_key, base_url=base_url)
+        # Initialize appropriate client based on endpoint_type
+        if endpoint_type == "azureopenai":
+            self.client = openai.AzureOpenAI(api_key=api_key, api_version=api_version or "2024-12-01-preview", azure_endpoint=base_url)
```

Comment:
let's just set this value "2024-12-01-preview" as default for api_version parameter?